File: hw/gudino/u3/hw29Singleton/SingletonJS/model/ConfigurationStock.js
class ConfigurationStock {
    constructor() {
        this.minimumStock = 10;
    }

    getMinimumStock() {
        return this.minimumStock;
    }

    setMinimumStock(value) {
        this.minimumStock = value;
    }
}

const instance = new ConfigurationStock();
export default instance;

File: hw/gudino/u3/hw29Singleton/SingletonJava/src/main/java/model/ConfigurationStock.java
package model;

public class ConfigurationStock {
    private static ConfigurationStock instance;
    private int minimumStock;

    private ConfigurationStock() {
        minimumStock = 10; // 
    }

    public static ConfigurationStock getInstance() {
        if (instance == null) {
            instance = new ConfigurationStock();
        }
        return instance;
    }

    public int getMinimumStock() {
        return minimumStock;
    }

    public void setMinimumStock(int minimumStock) {
        this.minimumStock = minimumStock;
    }
}
